src/ServicePulse.Host/vue/src/composables/serviceServiceControlUrls.ts

```diff
@@ -1,29 +1,36 @@
-import { ref } from "vue";
+import { type Ref, ref } from "vue";
 
-export const serviceControlUrl = ref(null);
-export const monitoringUrl = ref(null);
+export const serviceControlUrl = ref<string>();
+export const monitoringUrl = ref<string>();
 
 export function useIsMonitoringDisabled() {
-  return monitoringUrl.value === null || monitoringUrl.value === "" || monitoringUrl.value === "!";
+  return monitoringUrl.value === undefined || monitoringUrl.value === "" || monitoringUrl.value === "!";
 }
 
 export function useIsMonitoringEnabled() {
   return !useIsMonitoringDisabled();
 }
 
-export function useFetchFromServiceControl(suffix) {
+export function useFetchFromServiceControl(suffix: string) {
   return fetch(serviceControlUrl.value + suffix);
 }
 
-export function useFetchFromMonitoring(suffix) {
+export async function useTypedFetchFromServiceControl<T>(suffix: string): Promise<[Response, T]> {
+  const response = await fetch(serviceControlUrl.value + suffix);
+  const data = await response.json();
+
+  return [response, data];
+}
+
+export function useFetchFromMonitoring(suffix: string) {
   if (useIsMonitoringDisabled()) {
-    return Promise.resolve(null);
+    return Promise.resolve(undefined);
   }
   return fetch(monitoringUrl.value + suffix);
 }
 
-export function usePostToServiceControl(suffix, payload) {
-  const requestOptions = {
+export function usePostToServiceControl(suffix: string, payload: object | undefined = undefined) {
+  const requestOptions: RequestInit = {
     method: "POST",
   };
   if (payload !== undefined) {
```

Comment:
will also not want to add these headers if payload is null, so check should be if payload != null (i.e. it was wrong in the old version)